### PR TITLE
Create new role edit TSF

### PR DESCRIPTION
**Pull request by areyeslo**
# Main

- Create new role to edit TSF

# Other
- Creation of EDIT_TSF role in Keycloak 
- Change of MINE_EDIT to EDIT_TSF in PUT & POST method
- Change permission in Core-web for Add button from EDIT_PERMITS to EDIT_TSF
- Change permission in Core-web for Edit button from EDIT_REPORTS to EDIT_TSF
- Update permissions in tests
- Update snapshots
  

# How to test
- Login to Core-web with your IDIR and confirm you cannot Add/Edit a TSF.
- Login to minespace-web with BCEID and confirm that as proponent you can edit a TSF
- Assign the new role EDIT_TSF to your account in keycloak and login to Core-web with your IDIR account.
- Confirm you CAN Add/Edit a TSF

# Notes
https://bcmines.atlassian.net/browse/MDS-4094

### CODE REVIEW

File: services/core-api/tests/auth/test_expected_auth.py [hunk 1]
```diff
@@ -72,7 +72,7 @@
      (MineReportListResource, "post", [EDIT_REPORT, MINESPACE_PROPONENT]),
      (MineStatusXrefListResource, "get", [VIEW_ALL]),
      (MineTailingsStorageFacilityListResource, "get", [VIEW_ALL]),
-     (MineTailingsStorageFacilityListResource, "post", [MINE_EDIT]),
+     (MineTailingsStorageFacilityListResource, "post", [EDIT_TSF]),
```
Comment (BenHarrington) [inline services/core-api/tests/auth/test_expected_auth.py:76]: Should we also have another entry for the "put" given that it requires the same permissions?

**Review by BenHarrington — CHANGES_REQUESTED** [review]
One other thing that would be nice is to add the new role to the Production Keycloak, just so it is easier for Colin, Harp or whomever to add that role fast to select users.

Otherwise I've tested it and things are behaving as expected. Great work!

**Review by JustinMacaulay — COMMENTED** [review]
Looks good and my only comment was the same as bens so I removed it.

**Review by BenHarrington — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**